crates/ethcore/types/src/views/typed_transaction.rs

```diff
@@ -51,11 +53,15 @@ impl<'a> TypedTransactionView<'a> {
     /// }
     /// ```
     pub fn new(rlp: ViewRlp<'a>) -> TypedTransactionView<'a> {
-        TypedTransactionView { rlp: rlp }
+        let transaction_type = Self::extract_transaction_type(&rlp.rlp);
+        TypedTransactionView {
+            rlp: rlp,
+            transaction_type,
+        }
     }
 
-    pub fn transaction_type(&self) -> TypedTxId {
-        let ref rlp = self.rlp.rlp;
+    /// Extract transaction type from rlp bytes.
+    fn extract_transaction_type(rlp: &Rlp) -> TypedTxId {
```

Comment:
I would prefer:

```
let tx = rlp.data().unwrap();
let id = tx[0].try_into().unwrap();
```

to 

```
let tx = rlp.data().expect("unable to decode tx rlp");
let id = tx[0].try_into().expect("unable to decode tx type");
```

better for me is, returning `Option<TypedTxId>`

Reply:
Yep, it is better. Change is made.